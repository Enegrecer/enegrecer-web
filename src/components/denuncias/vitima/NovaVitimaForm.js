import React from 'react';
import CheckBox from '../../comum/checkbox';
import * as ConstantesCSS from '../ConstantesCss';
import Nome from '../../comum/nome';
import Genero from '../../comum/genero';
import Telefone from '../../comum/telefone';
import ComboboxRaca from './../../comum/comboboxRaca';
import Combobox from '../../comum/combobox';
import Data from '../../comum/data';

const NovaVitimaForm = () => (
  <div>
    <div className="row">
      <CheckBox
        state={'souAVitima'}
        divClasse={'col s12 m6 l6'}
        id={'souAVitima'}
        label={'Marque se você é a vítima'}
      />
    </div>
    <div className="row">
      <Nome id={'nome-vitima'} state={'nomeVitima'} />
      <Data
        state={'dataNascimentoVitima'}
        id={'dataNascimento'}
        label={'Data de Nascimento'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s12 m6 l6`}
      />
    </div>

    <div className="row">
      <Telefone
        divClasse={'input-field col s12 m6 l6'}
        id="telefone"
        state={'telefoneVitima'}
        label={'Telefone'}
      />
      <Combobox
        id={'cidade-vitima'}
        state={'cidadeVitima'}
        itens={['Belo Horizonte']}
        divClasse={'col s12 m6 l6'}
        label={'Cidade *'}
      />
    </div>

    <div className="row">
      <Genero
        state={'generoVitima'}
        id={'vitima-genero'}
        divClasse={'col s12 m6 l6'}
      />

      <ComboboxRaca
        state={'racaVitima'}
        id={'raca'}
        divClasse={'col s12 m6 l6'}
        somenteRacasVitima
      />
    </div>
  </div>
);

export default NovaVitimaForm;
