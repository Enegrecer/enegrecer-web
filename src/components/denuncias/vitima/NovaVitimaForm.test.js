import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import NovaVitimaForm from './NovaVitimaForm';


describe('NovaVitimaForm Component', () => {
  it('renderiza o componente sem erros', () => {
    const wrapper = shallow(<NovaVitimaForm />);
    expect(wrapper.exists()).toBe(true);
  });

  it('deve iniciar o state corretamente', () => {
    const wrapper = shallow(<NovaVitimaForm />)
    const expectedState = {
      pessoaIdentificada: false,
      souAVitima: false,
      conhecoAVitima: false,
      nome: '',
      genero: '',
      raca: '',
      dataNascimento: '',
      endereco: '',
      estado: '',
      telefone: '',
      email: '',
      naturalidade: '',
      caracteristicasVitima: '',
    };
    expect(wrapper.instance().state).toEqual(expectedState)
  });

  describe('quando o valor do campo for alterado', () => {
    const handleChangeSpy = spy();
    const wrapper = shallow(<NovaVitimaForm handleChange={handleChangeSpy} />);

    it('altera o valor correspondente no estado', () => {
      const eventMock = {
        target: {
          value: 'Coletivo nacional de Juventude Enegrecer é uma expressão do movimento social',
        },
      };

      wrapper.find('#nome-vitima').simulate('change', eventMock);
      expect(wrapper.state().nome).toEqual('Coletivo nacional de Juventude Enegrecer');
    });

    it('deve chamar o handleChange prop quando o component montar', () => {
      wrapper.instance().componentDidMount()
      expect(handleChangeSpy.called).toBeTruthy();
    })
  });
})
