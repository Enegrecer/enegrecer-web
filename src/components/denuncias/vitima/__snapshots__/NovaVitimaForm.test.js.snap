// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NovaVitimaForm Component deve possuir estrutura definida  1`] = `
<div>
  <div
    className="row"
  >
    <checkBox
      divClasse="col s12 m6 l6"
      id="souAVitima"
      label="Marque se você é a vítima"
      state="souAVitima"
    />
  </div>
  <div
    className="row"
  >
    <nome
      divClasse="col s12 m6 l6"
      id="nome-vitima"
      onChange={[Function]}
      state="nomeVitima"
      validacoes={Array []}
    />
    <Data
      divClasse="input-field col s12 m6 l6"
      id="dataNascimento"
      label="Data de Nascimento"
      state="dataNascimentoVitima"
    />
  </div>
  <div
    className="row"
  >
    <telefone
      divClasse="input-field col s12 m6 l6"
      id="telefone"
      label="Telefone"
      state="telefoneVitima"
    />
    <combobox
      divClasse="col s12 m6 l6"
      id="cidade-vitima"
      itens={
        Array [
          "Belo Horizonte",
        ]
      }
      label="Cidade *"
      state="cidadeVitima"
      valorPadrao=""
    />
  </div>
  <div
    className="row"
  >
    <genero
      divClasse="col s12 m6 l6"
      id="vitima-genero"
      state="generoVitima"
    />
    <comboRaca
      divClasse="col s12 m6 l6"
      id="raca"
      somenteRacasVitima={true}
      state="racaVitima"
    />
  </div>
</div>
`;
